model/godleyIcon.h

```diff
@@ -33,8 +33,24 @@
 namespace minsky
 {
   class GodleyTableEditor;
-
-  class GodleyIcon: public ItemT<GodleyIcon>
+  
+  // Godley variables broken out in a separate structure, as copying is non-default
+  struct GodleyVars
+  {
+  public:	    
+	  
+    GodleyVars() {}
+    virtual ~GodleyVars() {}
+    // copy operations not deleted to allow ItemT<GodleyIcon> to compile
+    GodleyVars(const GodleyVars& x) {} //{const_cast<GodleyVars&>(x).update();}
```

Comment:
This should all update() as well.

Reply:
I stiil don't understand why I need to call *this->update() or just update() in the copy constructor and not x.update()?

When I write     GodleyVars(const GodleyVars& x) {update();}

I get a compile warning:

model/godleyIcon.h: In copy constructor ‘minsky::GodleyVars::GodleyVars(const minsky::GodleyVars&)’:
model/godleyIcon.h:45:45: warning: pure virtual ‘virtual void minsky::GodleyVars::update()’ called from constructor
     GodleyVars(const GodleyVars& x) {update();}

and finally the build fails at the linking stage:

godleyIcon.o: In function `minsky::GodleyVars::GodleyVars(minsky::GodleyVars const&)':
/home/wd/minsky/model/godleyIcon.h:45: undefined reference to `minsky::GodleyVars::update()'
collect2: error: ld returned 1 exit status
Makefile:162: recipe for target 'gui-tk/minsky' failed
make: *** [gui-tk/minsky] Error 1

If you like, we can discuss it in our meeting at between 7 and 7.30 pm your time?